sdk/cosmos/src/clients/document_client.rs

```diff
@@ -85,9 +81,45 @@ impl DocumentClient {
         GetDocumentResponse::try_from(response).await
     }
 
+    /// replace a document in a collection
+    pub async fn replace_document<'a, T: Serialize>(
+        &self,
+        ctx: Context,
+        document: &'a T,
+        options: ReplaceDocumentOptions<'_>,
+    ) -> crate::Result<ReplaceDocumentResponse> {
+        let mut request = self.prepare_request_pipeline_with_document_name(http::Method::PUT);
+        let mut pipeline_context = PipelineContext::new(ctx, ResourceType::Databases.into());
```

Comment:
This context should be Documents.  #375 fixes this for GET, but obvs not for your new code.